pkg/providers/azure_keyvault.go

```diff
@@ -26,31 +35,55 @@ func NewAzureKeyVault(logger logging.Logger) (core.Provider, error) {
 		return nil, fmt.Errorf("cannot find KVAULT_NAME for azure key vault")
 	}
 
-	authorizer, err := kvauth.NewAuthorizerFromEnvironment()
+	var authorizer autorest.Authorizer
+	var err error
+	if os.Getenv("AZURE_CLI") == "1" {
```

Comment:
generally better to check for just "exists" no? (i.e. not null, string > 0).

Reply:
changed